src/components/Photography/PhotographyGridSection.tsx

```diff
@@ -3,148 +3,141 @@ import 'react-image-lightbox/style.css'
+    [4, 3, 2],
+    1,
   )
-
-  public UNSAFE_componentWillMount() {
-    this.setState({
-      lightboxImages: this.getLightboxImagesFromProps(this.props),
-    })
-  }
-
-  public render() {
-    const { datetime, images, slug = '/#' } = this.props
-    const {
-      isLightboxOpen,
-      lightboxSrc,
-      nextImageSrc,
-      prevImageSrc,
-    } = this.state
-    const sortedImages = _.sortBy(images, 'EXIF.DateTimeOriginal')
-    const lightboxImages = _.map(sortedImages, 'childImageSharp.sizes.src')
-
-    if (_.isEmpty(images) || _.isEmpty(lightboxImages)) {
-      return null
-    }
-
-    return (
-      <StyledPanel>
-        <PhotographySectionHeader datetime={datetime} href={slug} />
-        <ImageZoomGrid>
-          {sortedImages.map((image: any, imageIndex: number) => {
-            return (
-              <ImageZoomGridElement
-                key={image.id}
-                image={image}
-                onClick={() => this.clickImageElement(imageIndex)}
-              />
-            )
-          })}
-        </ImageZoomGrid>
-        {isLightboxOpen && lightboxSrc && (
-          <Lightbox
-            enableZoom={false}
-            mainSrc={lightboxSrc}
-            nextSrc={nextImageSrc}
-            prevSrc={prevImageSrc}
-            onCloseRequest={this.closeLightbox}
-            onMovePrevRequest={this.decrementLightboxIndex}
-            onMoveNextRequest={this.incrementLightboxIndex}
-          />
-        )}
-      </StyledPanel>
-    )
-  }
-
-  private decrementLightboxIndex = () => {
-    this.toggleLightbox({ index: this.state.index - 1, isLightboxOpen: true })
-  }
-
-  private incrementLightboxIndex = () => {
-    this.toggleLightbox({ index: this.state.index + 1, isLightboxOpen: true })
-  }
-
-  private closeLightbox = () => {
-    this.toggleLightbox({ ...this.state, isLightboxOpen: false })
-  }
-
-  private clickImageElement = (imageIndex: number) => {
-    this.toggleLightbox({
-      index: imageIndex,
-      isLightboxOpen: !this.state.isLightboxOpen,
-      lightboxImages: this.state.lightboxImages,
-    })
-  }
-
-  private toggleLightbox = ({
-    index = this.state.index,
-    isLightboxOpen = this.state.isLightboxOpen,
-    lightboxImages = this.state.lightboxImages,
-  }: ToggleLightboxOptions) => {
-    // prettier-ignore
-    this.setState({ // lgtm [js/react/inconsistent-state-update]
+  // Measure the width of the container element.
+  // @ts-ignore
+  const [ref, { width }] = useDimensions()
+
+  // Form a grid of stacked images, using the width & column calculations we got
+  // from the `useMedia` and `useDimensions` hooks above.
+  const heights: number[] = new Array(columns).fill(0) // Each column gets a height starting with zero
+  const gridItems = sortedImages.map((child, index) => {
+    const { childImageSharp } = child
+    const imageWidth = child.width || width / columns
+    const aspectRatio = _.get(childImageSharp, 'sizes.aspectRatio')
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 8 spaces but found 4. (@typescript-eslint/indent)](https://app.codacy.com/app/jdstuart/jesses.io/pullRequest?prid=4576429)